File: src/main/java/com/suifeng/web/italker/push/bean/db/PushHistory.java
package com.suifeng.web.italker.push.bean.db;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

/**
 * 消息推送
 */

@Entity
@Table(name = "TB_PUSH_HISTORY")
public class PushHistory {

    @Id
    @PrimaryKeyJoinColumn
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    @Column(updatable = false, nullable = false)
    private String id;

    //推送的具体实体，存储的都是JSON字符串
    //BLOB是比TEXT更多的一个大字段类型
    @Column(nullable = false, columnDefinition = "BLOB")
    private String entity;

    //推送的实体类型
    @Column(nullable = false)
    private int entityType;

    //接收者，不允许为空
    //一个接受者可以接收很多推送的消息
    //加载一条推送消息的时候 直接加载用户信息
    @JoinColumn(name = "receiverId")
    @ManyToOne(optional = false, fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private User receiver;
    @Column(nullable = false, updatable = false, insertable = false)
    private String receiverId;

    @JoinColumn(name = "senderId")
    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private User sender;
    @Column(updatable = false, insertable = false)
    private String senderId;

    //接收者当前状态下的设备推送Id
    @Column
    private String receiverPushId;

    @CreationTimestamp
    @Column(nullable = false)
    private LocalDateTime createAt = LocalDateTime.now();

    //定义为更新时间戳
    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updateAt = LocalDateTime.now();

    //消息到达的时间,可为空
    @Column(nullable = false)
    private LocalDateTime arriveAt;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getEntity() {
        return entity;
    }

    public void setEntity(String entity) {
        this.entity = entity;
    }

    public int getEntityType() {
        return entityType;
    }

    public void setEntityType(int entityType) {
        this.entityType = entityType;
    }

    public User getReceiver() {
        return receiver;
    }

    public void setReceiver(User receiver) {
        this.receiver = receiver;
    }

    public String getReceiverId() {
        return receiverId;
    }

    public void setReceiverId(String receiverId) {
        this.receiverId = receiverId;
    }

    public User getSender() {
        return sender;
    }

    public void setSender(User sender) {
        this.sender = sender;
    }

    public String getSenderId() {
        return senderId;
    }

    public void setSenderId(String senderId) {
        this.senderId = senderId;
    }

    public String getReceiverPushId() {
        return receiverPushId;
    }

    public void setReceiverPushId(String receiverPushId) {
        this.receiverPushId = receiverPushId;
    }

    public LocalDateTime getCreateAt() {
        return createAt;
    }

    public void setCreateAt(LocalDateTime createAt) {
        this.createAt = createAt;
    }

    public LocalDateTime getUpdateAt() {
        return updateAt;
    }

    public void setUpdateAt(LocalDateTime updateAt) {
        this.updateAt = updateAt;
    }

    public LocalDateTime getArriveAt() {
        return arriveAt;
    }

    public void setArriveAt(LocalDateTime arriveAt) {
        this.arriveAt = arriveAt;
    }
}

File: src/main/java/com/suifeng/web/italker/push/bean/db/PushHistory.kt
package com.suifeng.web.italker.push.bean.db

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.UpdateTimestamp

import javax.persistence.*
import java.time.LocalDateTime

/**
 * 消息推送
 */

@Entity
@Table(name = "TB_PUSH_HISTORY")
class PushHistory {

    @Id
    @PrimaryKeyJoinColumn
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    @Column(updatable = false, nullable = false)
    var id: String? = null

    //推送的具体实体，存储的都是JSON字符串
    //BLOB是比TEXT更多的一个大字段类型
    @Column(nullable = false, columnDefinition = "BLOB")
    var entity: String? = null

    //推送的实体类型
    @Column(nullable = false)
    var entityType: Int = 0

    //接收者，不允许为空
    //一个接受者可以接收很多推送的消息
    //加载一条推送消息的时候 直接加载用户信息
    @JoinColumn(name = "receiverId")
    @ManyToOne(optional = false, fetch = FetchType.EAGER, cascade = [(CascadeType.ALL)])
    var receiver: User? = null
    @Column(nullable = false, updatable = false, insertable = false)
    var receiverId: String? = null

    @JoinColumn(name = "senderId")
    @ManyToOne(fetch = FetchType.EAGER, cascade = [(CascadeType.ALL)])
    var sender: User? = null
    @Column(updatable = false, insertable = false)
    var senderId: String? = null

    //接收者当前状态下的设备推送Id
    @Column
    var receiverPushId: String? = null

    @CreationTimestamp
    @Column(nullable = false)
    var createAt = LocalDateTime.now()

    //定义为更新时间戳
    @UpdateTimestamp
    @Column(nullable = false)
    var updateAt = LocalDateTime.now()

    //消息到达的时间,可为空
    @Column(nullable = false)
    var arriveAt: LocalDateTime? = null
}
